File: Actors.py
__author__ = 'b0ggyb33'
from World import positions,directions

class Actor(object):
    def __init__(self,limits,initialPosition=None):
        self.limits=limits
        if initialPosition == directions['LEFT']:
            self.position = self.limits[0]
        elif initialPosition == directions['RIGHT']:
            self.position = self.limits[1]
        else:
            self.position=int((self.limits[0]+self.limits[1]) / 2)
        self.live=True

    def move(self,direction):
        newPosition = self.position+direction
        if self.limits[0] <= newPosition <= self.limits[1]:
            self.position = newPosition
            return True
        else:
            return False

class Ball(Actor):
    def __init__(self,limits,initialPosition=None,track=0):
        super(Ball,self).__init__(limits,initialPosition)
        if initialPosition == directions['RIGHT']:
            self.velocity = directions['LEFT']
        else:
            self.velocity = directions['RIGHT']

        self.track=track
        self.hasBeenScored=True

    def atLimit(self):
        if self.position in self.limits:
            return True
        else:
            return False

    def update(self):
        self.hasBeenScored=False
        self.live = self.move(self.velocity)

    def changeDirection(self):
        if self.velocity == directions['LEFT']:
            self.velocity = directions['RIGHT']
        else:
            self.velocity = directions['LEFT']


class MrGameAndWatch(Actor):
    def __init__(self,limits,initialPosition=None):
        super(MrGameAndWatch,self).__init__(limits,initialPosition)

    def update(self):
        pass

File: src/Actors.c
#include <pebble.h>
#include "src/Actors.h"

void initialise_MisterGameAndWatch(MrGameAndWatch* object)
{
  object->lowerLimit = 0;
  object->upperLimit = 2;
  object->live=1;
  
  object->position = 1; //centre position
}

int8_t move_MisterGameAndWatch(MrGameAndWatch* object,int8_t direction)
{
  int8_t newPosition = object->position + direction;
  if ( (object->lowerLimit <= newPosition) && (newPosition <= object->upperLimit))
  {
    object->position = newPosition;
    return 1;
  }
  return 0;
}

//BALL CODE

void initialise_Ball(Ball* object,int8_t lowerLimit,int8_t upperLimit, int8_t initialPosition, int8_t track)
{
  object->lowerLimit = lowerLimit;
  object->upperLimit = upperLimit;
  object->live=1;
  object->hasBeenScored = 1;
  object->track = track;
  
  if (initialPosition == DIRECTION_LEFT)
  {
    object->position = object->lowerLimit;
  }
  else if (initialPosition == DIRECTION_RIGHT)
  {
    object->position = object->upperLimit;
  }
  else
  {
    object->position = 1;
  }
  
  if (initialPosition == DIRECTION_RIGHT)
  {
    object->velocity = DIRECTION_LEFT;
  }
  else if (initialPosition == DIRECTION_LEFT)
  {
    object->velocity = DIRECTION_RIGHT;
  }
  else
  {
    object->velocity = 0;
  }
}

int8_t move_Ball(Ball* object,int8_t direction)
{
  char newPosition = object->position + direction;
  if ( (object->lowerLimit <= newPosition) && (newPosition <= object->upperLimit))
  {
    object->position = newPosition;
    return 1;
  }
  return 0;
}

int8_t atLimit(Ball* object)
{
  if (object->position == object->lowerLimit || object->position == object->upperLimit)
  {
    return 1;
  }
  else
  {
    return 0;
  }
}

void update(Ball* object)
{
  object->hasBeenScored=0;
  object->live = move_Ball(object, object->velocity);
}

void changeDirection(Ball* object)
{
  if (object->velocity == DIRECTION_LEFT)
  {
    object->velocity = DIRECTION_RIGHT;
  }
  else
    object->velocity = DIRECTION_LEFT;
}
